### PR TITLE
add queries for get article function

### PR DESCRIPTION
**Pull request by rorybush**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -322,3 +322,67 @@ describe("GET /api/users", () => {
...
 });
+
+describe("GET /api/articles (queries)", () => {
+  describe("topic type query", () => {
+    test("return status 200 and filters the articles by topic", () => {
+      return request(app)
+        .get("/api/articles?topic=cats")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
```
Comment (B7Bprog) [inline __tests__/app.test.js:333]: Please check the length of the 'articles' array as well.

File: __tests__/app.test.js [hunk 2]
```diff
@@ -322,3 +322,67 @@ describe("GET /api/users", () => {
...
+          articles.forEach((article) => {
+            expect(article).toEqual(
+              expect.objectContaining({
+                topic: "cats",
+              })
+            );
+          });
+        });
+    });
+    test("returns status 400 if an invalid topic has been entered", () => {
```
Comment (B7Bprog) [inline __tests__/app.test.js:343]: This needs to be a 404, 'not found', please amend it.

File: __tests__/app.test.js [hunk 3]
```diff
@@ -322,3 +322,67 @@ describe("GET /api/users", () => {
...
+    });
+    test("200 - sort by author in ascending order", () => {
+      return request(app)
+        .get("/api/articles?sort_by=author&order=ASC")
+        .expect(200)
+        .then(({ body }) => {
+          const { articles } = body;
+          expect(articles).toBeSortedBy("author");
+        });
+    });
```
Comment (B7Bprog) [inline __tests__/app.test.js:370]: Please write another test where the topic exists, but there are no articles. E.g. ?topic=paper . This should give a status 200 and an empty array of articles.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Rory, please see my comments and amend your code. When you're done, please send us another pr with it. 👍

**Review by roisinoreilly — APPROVED** [review]
nicely refactored, merge when ready

**Pull request merged**